client/ayon_core/plugins/load/delete_old_versions.py

```diff
@@ -1,501 +1,420 @@
+                    collections.remove(final_col)
+
+                elif os.path.exists(file_path):
+                    size += os.path.getsize(file_path)
+
+                    if delete:
+                        os.remove(file_path)
+                        self.log.debug("Removed file: {}".format(file_path))
+                else:
+                    self.log.debug(
+                        "File was not found: {}".format(file_path)
+                    )
+
+        # Delete as much as possible parent folders
+        if not delete:
+            return size
+
+        for dir_path in dir_paths.values():
+            while True:
+                if not os.path.exists(dir_path):
+                    dir_path = os.path.dirname(dir_path)
+                    continue
+
+                if len(os.listdir(dir_path)) != 0:
+                    break
+
+                self.log.debug("Removed folder: {}".format(dir_path))
+                os.rmdir(dir_path)
+
+        return size
+
+    def message(self, text):
+        msgBox = QtWidgets.QMessageBox()
+        msgBox.setText(text)
+        msgBox.setStyleSheet(style.load_stylesheet())
+        msgBox.setWindowFlags(
+            msgBox.windowFlags() | QtCore.Qt.FramelessWindowHint
+        )
+        msgBox.exec_()
+
+    def get_data(self, context, versions_count):
+        product_entity = context["product"]
+        folder_entity = context["folder"]
+        project_name = context["project"]["name"]
+        anatomy = Anatomy(project_name, project_entity=context["project"])
+
+        versions = list(ayon_api.get_versions(
+            project_name,
+            product_ids=[product_entity["id"]],
+            active=None,
+            hero=False
+        ))
+        self.log.debug(
+            "Version Number ({})".format(len(versions))
+        )
+        versions_by_parent = collections.defaultdict(list)
+        for ent in versions:
+            versions_by_parent[ent["productId"]].append(ent)
+
+        def sort_func(ent):
+            return int(ent["version"])
+
+        all_last_versions = []
+        for _parent_id, _versions in versions_by_parent.items():
+            for idx, version in enumerate(
+                sorted(_versions, key=sort_func, reverse=True)
+            ):
+                if idx >= versions_count:
+                    break
+                all_last_versions.append(version)
+
+        self.log.debug("Collected versions ({})".format(len(versions)))
+
+        # Filter latest versions
+        for version in all_last_versions:
+            versions.remove(version)
+
+        # Update versions_by_parent without filtered versions
+        versions_by_parent = collections.defaultdict(list)
+        for ent in versions:
+            versions_by_parent[ent["productId"]].append(ent)
+
+        # Filter already deleted versions
+        for version in versions:
+            if version["active"] or "deleted" in version["tags"]:
```

Comment:
My fault, this probably should be only check for tags...
```suggestion
            if "deleted" in version["tags"]:
```